### PR TITLE
Set random default password for admin user

### PR DESCRIPTION
**Pull request by fwilhe**
Secure Jenkins instance by default by setting a random password for the admin user. This is done only if Jenkins is not already secured (in jenkins home volume). The password is written to stdout so it is reachable via docker logs.

### CODE REVIEW

**Comment by fwilhe** [comment]
Manually tested that when I boot up, set a new password, do stop remove start my set password is still valid.

File: cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Notes on running CX Server in Production
+
+Please be aware that running a Jenkins in production securly is not a trivial task.
+
+Some recommendations:
+
+The `latest` tag of CX Server Docker images are rebuilt on a regular basis.
+To benefit from updated Jenkins core and plugin versions, you should stop, remove and restart your CX Server instance on a regular basis.
+Part of this procedure should be to take backups (see `backup` and `restore` commands of `cx-server`).
+The commands perform an update of CX Server are:
```
Comment (stippi2) [inline cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md]:
```suggestion
The commands to perform an update of the CX Server are:
```

File: cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Notes on running CX Server in Production
+
+Please be aware that running a Jenkins in production securly is not a trivial task.
+
+Some recommendations:
+
+The `latest` tag of CX Server Docker images are rebuilt on a regular basis.
```
Comment (stippi2) [inline cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md]:
```suggestion
The `latest` tag of the CX Server Docker images are rebuilt on a regular basis.
```

File: cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+To benefit from updated Jenkins core and plugin versions, you should stop, remove and restart your CX Server instance on a regular basis.
```
Comment (stippi2) [inline cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md]:
```suggestion
To benefit from updated Jenkins core and plugin versions, you should stop, remove, and restart your CX Server instance on a regular basis.
```

**Comment by daniel-kurzynski** [comment]
Windows ...
image

**Comment by fwilhe** [comment]
> Windows ...
> image

so how do you grep in cmd.exe?

seriously, with wsl 2 on the horizon I would compromise here. Maybe rephrase the message for windows users, but not more.

**Comment by daniel-kurzynski** [comment]
My comments:
- It looks like I had still an old instance running. After executing cx-server start, it was not secured. I guess I have to first run cx-server remove!?
- grep does not work on windows
- From the output it is not really clear what the username
- Maybe you could make it more clear that I should run the command in some terminal, especially you could highlight the command by using quotes ...

**Comment by daniel-kurzynski** [comment]
What about cx-server get-password

**Comment by daniel-kurzynski** [comment]
After cx-server remove it is still not secured...

**Comment by fwilhe** [comment]
> After cx-server remove it is still not secured...

How did you setup your environment?

I'm still struggling with developer mode, but basically you need to ensure that you re-build the PR branch and don't overwrite your images. I've started a registry locally and pushed my images there and set it in server.cfg.

**Comment by daniel-kurzynski** [comment]
Well I just removed every pull from the code before running docker build

**Comment by fwilhe** [comment]
> After cx-server remove it is still not secured...

I could not reproduce such behaviour. I've added debug logging to Jenkins log, if you can reproduce it would be interesting to see that. Let's see on monday.

**Comment by fwilhe** [comment]
I've added some happy-path testing to travis. Please review.

@SAP/jenkins-library-team

File: cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Notes on running CX Server in Production
+
+Please be aware that running a Jenkins in production securly is not a trivial task.
```
Comment (stippi2) [inline cx-server-companion/life-cycle-scripts/CX-SERVER-SECURITY.md]:
```suggestion
Please be aware that running a Jenkins in production securely is not a trivial task.
```

**Comment by fwilhe** [comment]
> Looks good to me still. I've wondered whether there should be another command "update" to the script, which combines "stop + remove + start" into one action. Could of course be the scope of another PR.

Yes, this is out of scope. Also we have discussed this in the past, but IIRC it was not as trivial as it sounds.
